galpy/potential/TwoPowerSphericalPotential.py

```diff
@@ -217,10 +219,244 @@ def _mass(self,R,z=0.,t=0.):
         HISTORY:
            2014-04-01 - Written - Erkal (IoA)
         """
-        if z is None: r= R
-        else: r= numpy.sqrt(R**2.+z**2.)
+        r= R if z is None else numpy.sqrt(R**2.+z**2.)
         return (r/self.a)**(3.-self.alpha)/(3.-self.alpha)*special.hyp2f1(3.-self.alpha,-self.alpha+self.beta,4.-self.alpha,-r/self.a)
 
+class DehnenSphericalPotential(TwoPowerSphericalPotential):
+    """Class that implements the Dehnen Spherical Potential from Dehnen 1993
+
+    https://doi.org/10.1093/mnras/265.1.250
+
+    .. math::
+
+          \\rho(r) = \\frac{\\mathrm{amp}(3-\\alpha)}{4\\,\\pi\\,a^3}\\,\\frac{1}{(r/a)^{\\alpha}\\,(1+r/a)^{4-\\alpha}}
+    """
+
+    def __init__(self,amp=1.,a=1.,alpha=1.5,normalize=False,ro=None,vo=None):
+        """
+        NAME:
+
+           __init__
+
+        PURPOSE:
+
+           initialize a Dehnen Spherical Potential
+
+           note that the amplitude does NOT match that of Dehnen 1992.
+           to match amplitude, use amp * (3 - alpha).
+
+        INPUT:
+
+           amp - amplitude to be applied to the potential (default: 1); can be a Quantity with units of mass or Gxmass will be rescaled to amp*(3-alpha) (except for Jaffe, which is already scaled)
+
+           a - scale radius (can be Quantity)
+
+           alpha - inner power, restricted to [0, 3)
+
+           normalize - if True, normalize such that vc(1.,0.)=1., or, if given as a number, such that the force is this fraction of the force necessary to make vc(1.,0.)=1.
+
+           ro=, vo= distance and velocity scales for translation into internal units (default from configuration file)
+
+        OUTPUT:
+
+           (none)
+
+        HISTORY:
+
+           2019-10-07 - Started - Starkman (UofT)
+
+        """
+        if (alpha < 0.) or (alpha >= 3.):
+            raise ValueError('alpha must be [0, 3)')
+        # instantiate
+        TwoPowerSphericalPotential.__init__(
+            self,amp=amp,a=a,alpha=alpha,beta=4,
+            normalize=normalize,ro=ro,vo=vo
+        )
+        # make integer-self and protect subclasses
+        if ((self.__class__ == DehnenSphericalPotential) &
+            (alpha == round(alpha))):
+            self.integerSelf= TwoPowerIntegerSphericalPotential(
+                amp=1.,a=a,alpha=round(alpha),beta=4,
+                normalize=False,ro=None,vo=None)
+        else:
+            self.integerSelf= None
+        # set properties
+        self.hasC= True
+        self.hasC_dxdv= True
+        # self._nemo_accname= 'Dehnen'
```

Comment:
not sure if it Nemo compatible. How should I check?

Reply:
You should add a test to ``tests/test_nemo.py`` similar to the tests that are already in that file. These don't run on travis (like the AMUSE ones, because NEMO is too difficult to install on travis), but you should be able to run them on the server if you get NEMO to work there using [these instructions](http://astro.utoronto.ca/~bovy/group/compute_servers.html#nemo)

Reply:
I followed the instructions but all the Nemo tests (for all the potentials, not only dehnen) are failing. I imagine that means my NEMO is not working correctly.